identity/registry/registry_contract.go

```diff
@@ -18,51 +18,35 @@
 package registry
 
 import (
-	"github.com/mysteriumnetwork/payments/registry"
-	"github.com/mysteriumnetwork/payments/registry/generated"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
-	"github.com/ethereum/go-ethereum/accounts/keystore"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/payments/registry/generated"
 )
 
-// IdentityRegistry checks whenever given identity is registered
-type IdentityRegistry interface {
-	IsRegistered(identity common.Address) (bool, error)
-}
-
-// RegistrationDataProvider provides registration information for given identity required to register it on blockchain
-type RegistrationDataProvider interface {
-	ProvideRegistrationData(identity common.Address) (*registry.RegistrationData, error)
-}
-
-type keystoreRegistrationDataProvider struct {
-	ks *keystore.KeyStore
-}
-
-func (kpg *keystoreRegistrationDataProvider) ProvideRegistrationData(identity common.Address) (*registry.RegistrationData, error) {
-	identityHolder := registry.FromKeystore(kpg.ks, identity)
-
-	return registry.CreateRegistrationData(identityHolder)
-}
-
-// NewRegistrationDataProvider creates registration data provider backed up by identity which is managed by keystore
-func NewRegistrationDataProvider(ks *keystore.KeyStore) RegistrationDataProvider {
-	return &keystoreRegistrationDataProvider{
-		ks: ks,
-	}
-}
-
 // NewIdentityRegistry creates identity registry service which uses blockchain for information
-func NewIdentityRegistry(contractCaller bind.ContractCaller, registryAddress common.Address) (IdentityRegistry, error) {
+func NewIdentityRegistry(contractCaller bind.ContractCaller, registryAddress common.Address) (*contractRegistry, error) {
```

Comment:
I would leave IdentityRegistry as is, since the only way identities can be registered is through contract.
Factory method also is called NewIdentityRegistry not NewContractRegistry.

Reply:
We dont return interface type, but type of implementation. This is why I renamed

Reply:
Renamed to `NewIdentityRegistryContract()`